src/segy/file.py

```diff
@@ -196,35 +198,59 @@ def _parse_binary_header(self) -> None:
     @property
     def data(self) -> AbstractIndexer:
         """Way to access the file to fetch trace data only."""
+        transforms = [
+            TransformFactory.create("byte_swap", Endianness.LITTLE),
+        ]
+
+        if self.spec.trace.data_descriptor.format == ScalarType.IBM32:
+            transforms.append(TransformFactory.create("ibm_float", "to_ieee"))
+
```

Comment:
Are the property methods where we should handle the decoding for all of the accessors?

Reply:
no idea, that's the part we need to figure it out. i currently left everything coupled like before. the high level specs are:
1. read bytes
2. decode
3. transform

so how we do it is up to you like we discussed earlier. let's try to find a clean way